src/DGtal/helpers/StdDefs.cpp

```diff
@@ -0,0 +1,68 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+/**
+ * @file StdDefs.cpp
+ * @author David Coeurjolly (\c david.coeurjolly@liris.cnrs.frx )
+ *
+ * @date 2016/01/19
+ *
+ * This file is part of the DGtal library.
+ */
+
+#include "DGtal/helpers/StdDefs.h"
+
+
+//////////////////
+/// Z2i Explicit instanciation
+//////////////////
+template class DGtal::SpaceND<2, DGtal::Z2i::Integer>;
+template class DGtal::PointVector<2, DGtal::Z2i::Integer>;
+template class DGtal::KhalimskySpaceND< 2, DGtal::Z2i::Integer > ;
+template class DGtal::HyperRectDomain< DGtal::Z2i::Space > ; 
+template class DGtal::MetricAdjacency< DGtal::Z2i::Space, 1> ;
+template class DGtal::MetricAdjacency< DGtal::Z2i::Space, 2> ;
+template class DGtal::DigitalTopology< DGtal::Z2i::Adj4, DGtal::Z2i::Adj8 > ;
+template class DGtal::DigitalTopology< DGtal::Z2i::Adj8, DGtal::Z2i::Adj4 > ;
+template class DGtal::DigitalSetByAssociativeContainer<DGtal::Z2i::Domain,
+                                                       std::unordered_set< typename DGtal::Z2i::Domain::Point> >;
+//template class DGtal::Object<DGtal::Z2i::DT4_8, DGtal::Z2i::DigitalSet> ;
+//template class DGtal::Object<DGtal::Z2i::DT8_4, DGtal::Z2i::DigitalSet> ;
+template class DGtal::GridCurve<DGtal::Z2i::K2> ;
+template class DGtal::ExactPredicateLpSeparableMetric<DGtal::Z2i::Space,2> ;
+template class DGtal::ExactPredicateLpSeparableMetric<DGtal::Z2i::Space,1> ;
+template class DGtal::ExactPredicateLpPowerSeparableMetric<DGtal::Z2i::Space,2> ;
+template class DGtal::ExactPredicateLpPowerSeparableMetric<DGtal::Z2i::Space,1> ;
```

Comment:
I see but it seems that a typedef is not enough to explicitly instantiate a template and since `SpaceND` do not have any member ...
However, `HyperRectDomain` does have point members, so it is probably not needed.